### PR TITLE
Add content/en/hypervisor

### PR DESCRIPTION
**Pull request by KiranSatyaRaj**
adding the term hypervisor to the glossary

Signed-off-by: Kiran Satya Raj <90622705+KiranSatyaRaj@users.noreply.github.com>

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

### CODE REVIEW

File: content/en/hypervisor.md
Comment (NitishKumar06) [inline content/en/hypervisor.md:11]:
```suggestion
by taking the advantage of [bare metal machine](/bare-metal-machine/) resources
```

**Comment by KiranSatyaRaj** [comment]
Thanks for suggesting the changes @NitishKumar06, they were relevant so I've committed them.

File: content/en/hypervisor.md
Comment (iamNoah1) [inline content/en/hypervisor.md:21]:
```suggestion
Devoting all the resources of that server to just a single application was not the most effective way.
```

File: content/en/hypervisor.md
Comment (iamNoah1) [inline content/en/hypervisor.md:22]: The content of the rest of this section is not very accurate, or easy to follow. The aspect of having wasted a whole powerful server on just one app is an important aspect that can be followed up on. For example just saying, that a server could hold more apps, but then you have isolation and noisy neighbors problems (one highly resource-consuming app has an impact on the other ones). 
Another problem of the world before virtualization is setup costs. VMs are more quickly set up than bare metal. This is one step to closing the gap between dev and ops.

File: content/en/hypervisor.md
Comment (iamNoah1) [inline content/en/hypervisor.md:30]:
```suggestion

```

File: content/en/hypervisor.md
Comment (iamNoah1) [inline content/en/hypervisor.md:29]:
```suggestion
The Hypervisor is used to create and allocate resources to virtual machines until we reach the saturating performance of the physical machines.
```

File: content/en/hypervisor.md
Comment (iamNoah1) [inline content/en/hypervisor.md:31]:
```suggestion
```

as it does not really add value to the definition

File: content/en/hypervisor.md
Comment (iamNoah1) [inline content/en/hypervisor.md:32]:
```suggestion

```

File: content/en/hypervisor.md
Comment (iamNoah1) [inline content/en/hypervisor.md:34]: I would try to focus more on the setup speed and the degree of isolation between apps than on migrating for the rest of this section.

**Review by iamNoah1 — CHANGES_REQUESTED** [review]
@KiranSatyaRaj Thanks for the contribution. A good start, but I have some change requests and remarks :)

File: content/en/hypervisor.md
Comment (KiranSatyaRaj) [reply content/en/hypervisor.md:14]: yeah, you're right, knowing its functionality is more important than types so I'll add some more information and I'll get back to you with that.

**Comment by NitishKumar06** [comment]
@iamNoah1 I'm afraid if the website contains some kind of checks where the build gets failed due to large number of commits raised. I have no idea till now if such check exists.  In order to keep on the safer side, It could be better if @KiranSatyaRaj squash all the commits. WDYT @iamNoah1 ?

**Comment by seokho-son** [comment]
Thanks @NitishKumar06 

large number of commits does not incur build failures. But, since meaningless commits are not desired, squashing commits in a PR is desired as @NitishKumar06 suggested. 
By the way, contributor doesn't have to keep squashing or amending commits for every PR updates. The squashing can be done by contributor just before this PR is ready to be merged. 

Anyway, an approver will use `squash and merge` method to merge this PR.

File: content/en/hypervisor.md
Comment (KiranSatyaRaj) [reply content/en/hypervisor.md:23]: I'm thinking these points could go into this section " acquiring a physical machine takes time, it requires a team of engineers to monitor them, whereas VMs are more quickly set up than bare metal", what do you think? @iamNoah1 and is it because a hypervisor takes away the low-level complexities (like abstraction layer over the physical hardware) so that it is easier to set up a VM whether they are of the dev team or Ops team?

**Review by iamNoah1 — COMMENTED** [review]
@KiranSatyaRaj please let me know whenever you have incorporated or considered the suggested feedback and if we should took a look again :)

**Comment by KiranSatyaRaj** [comment]
> @KiranSatyaRaj please let me know whenever you have incorporated or considered the suggested feedback and if we should took a look again :)

Sure, there were a few points I mentioned in the discussion regarding the term I just wanted to check whether they were neutral and straightforward, that's fine I'll work my way through it and get back to you with the applied changes as soon as possible.

File: content/en/hypervisor.md
Comment (seokho-son) [inline content/en/hypervisor.md:10]: @KiranSatyaRaj 
I think `software application` may not correct technically.
In general, there are two types of hypervisor. 
 1) bare metal/native hypervisor that runs on bare metal computer directly (e.g., KVM, Xen, ...)
 2) hosted hypervisor that runs on OS as an application (e.g., Virtual box)

So, if we say `A hypervisor is a software application`, readers may be confused with the second type of hypervisor.

**Comment by KiranSatyaRaj** [comment]
Hey @iamNoah1 I made, can you please go through it and let me know what more can be done to make it better.

**Review by iamNoah1 — APPROVED** [review]
Thanks, @KiranSatyaRaj for your contribution and for your patience with all that back and forth. I think this is a good start so we will merge it as a draft. Except @CathPag objects :)

**Review by CathPag — APPROVED** [review]
LGTM for draft

**Pull request merged**
